### PR TITLE
Provide ability to pass in file sha in GithubApi#update_contents

### PR DESCRIPTION
**Pull request by camertron**
@lumoslabs/platform

### CODE REVIEW

File: txgh/lib/txgh/github_api.rb [hunk 1]
```diff
@@ -34,17 +34,21 @@ def create_ref(branch, sha)
...
+        file_sha = file_params.fetch(:sha) do
+          begin
+            client.contents(repo_name, { path: path, ref: branch })[:sha]
+          rescue Octokit::NotFound
+            nil
+          end
         end
 
-        current_sha = file ? file[:sha] : '0' * 40
+        current_sha = file_sha || '0' * 40
```
Comment (jdoconnor) [inline txgh/lib/txgh/github_api.rb:54]: comment to explain why we're making up a fake sha, please

File: txgh/lib/txgh/github_api.rb [hunk 2]
```diff
@@ -34,17 +34,21 @@ def create_ref(branch, sha)
       client.create_ref(repo_name, branch, sha) rescue false
     end
 
-    def update_contents(branch, content_map, message)
-      content_map.each do |path, new_contents|
+    def update_contents(branch, content_list, message)
+      content_list.each do |file_params|
+        path = file_params.fetch(:path)
+        new_contents = file_params.fetch(:contents)
```
Comment (jdoconnor) [inline txgh/lib/txgh/github_api.rb:40]: is this being used?

**Review by jdoconnor — CHANGES_REQUESTED** [review]

File: txgh/lib/txgh/github_api.rb (hunk 2 above)
Comment (camertron) [reply txgh/lib/txgh/github_api.rb:40]: Yeah, it's used by the [`ResourceCommitter`](https://github.com/lumoslabs/txgh/blob/055eccdacc3c335e4a53086ed68ecd2f771c23c7/txgh/lib/txgh/resource_committer.rb#L21).

File: txgh/lib/txgh/github_api.rb (hunk 1 above)
Comment (camertron) [reply txgh/lib/txgh/github_api.rb:54]: It's the SHA you pass to github to make it create a file instead of update one. Comment added!

**Pull request merged**